x/oracle/keeper/grpc_query_test.go

```diff
@@ -137,8 +140,123 @@ func (s *IntegrationTestSuite) TestQuerier_AggregateVotes() {
+	resExchangeRate, err := s.queryClient.ExchangeRates(s.ctx.Context(), &types.QueryExchangeRatesRequest{
+		Denom: " ",
+	})
+	s.Require().Nil(resExchangeRate)
+	s.Require().ErrorContains(err, "unknown denom")
+}
+
+func (s *IntegrationTestSuite) TestQuerier_AggregatePrevoteInvalidValAddr() {
+	resExchangeRate, err := s.queryClient.AggregatePrevote(s.ctx.Context(), &types.QueryAggregatePrevoteRequest{
+		ValidatorAddr: "valaddrInvalid",
+	})
+	s.Require().Nil(resExchangeRate)
+	s.Require().ErrorContains(err, "decoding bech32 failed")
+}
+
+func (s *IntegrationTestSuite) TestQuerier_AggregatePrevotesAppendVotes() {
+	s.app.OracleKeeper.SetAggregateExchangeRatePrevote(s.ctx, valAddr, types.NewAggregateExchangeRatePrevote(
+		types.AggregateVoteHash{},
+		valAddr,
+		uint64(s.ctx.BlockHeight()),
+	))
+
+	_, err := s.queryClient.AggregatePrevotes(s.ctx.Context(), &types.QueryAggregatePrevotesRequest{})
+	s.Require().Nil(err)
+}
+
+func (s *IntegrationTestSuite) TestQuerier_AggregateVotesAppendVotes() {
+	s.app.OracleKeeper.SetAggregateExchangeRateVote(s.ctx, valAddr, types.NewAggregateExchangeRateVote(
+		types.DefaultGenesisState().ExchangeRates,
+		valAddr,
+	))
+
+	_, err := s.queryClient.AggregateVotes(s.ctx.Context(), &types.QueryAggregateVotesRequest{})
+	s.Require().Nil(err)
+}
+
+func TestEmptyRequest(t *testing.T) {
+	q := keeper.NewQuerier(keeper.Keeper{})
+	emptyRequestErrorMsg := "empty request"
+
+	resParams, err := q.Params(context.Background(), nil)
+	require.Nil(t, resParams)
+	require.ErrorContains(t, err, emptyRequestErrorMsg)
+
+	resExchangeRate, err := q.ExchangeRates(context.Background(), nil)
+	require.Nil(t, resExchangeRate)
+	require.ErrorContains(t, err, emptyRequestErrorMsg)
+
+	resActiveExchangeRates, err := q.ActiveExchangeRates(context.Background(), nil)
+	require.Nil(t, resActiveExchangeRates)
+	require.ErrorContains(t, err, emptyRequestErrorMsg)
+
+	resFeederDelegation, err := q.FeederDelegation(context.Background(), nil)
+	require.Nil(t, resFeederDelegation)
+	require.ErrorContains(t, err, emptyRequestErrorMsg)
+
+	resMissCounter, err := q.MissCounter(context.Background(), nil)
+	require.Nil(t, resMissCounter)
+	require.ErrorContains(t, err, emptyRequestErrorMsg)
+
+	resAggregatePrevote, err := q.AggregatePrevote(context.Background(), nil)
+	require.Nil(t, resAggregatePrevote)
+	require.ErrorContains(t, err, emptyRequestErrorMsg)
+
+	resAggregatePrevotes, err := q.AggregatePrevotes(context.Background(), nil)
+	require.Nil(t, resAggregatePrevotes)
+	require.ErrorContains(t, err, emptyRequestErrorMsg)
+
+	resAggregateVote, err := q.AggregateVote(context.Background(), nil)
+	require.Nil(t, resAggregateVote)
+	require.ErrorContains(t, err, emptyRequestErrorMsg)
+
+	resAggregateVotes, err := q.AggregateVotes(context.Background(), nil)
+	require.Nil(t, resAggregateVotes)
+	require.ErrorContains(t, err, emptyRequestErrorMsg)
+}
+
+func TestInvalidBechAddress(t *testing.T) {
```

Comment:
Same here, we should use the test suite in each method: 
```suggestion
func (s *IntegrationTestSuite) TestInvalidBechAddress() {
```